app/invocation/invocation_action_card.tsx

```diff
@@ -517,6 +531,62 @@ export default class InvocationActionCardComponent extends React.Component<Props
       });
   }
 
+  private renderOutputDirectories(actionsResult: build.bazel.remote.execution.v2.ActionResult) {
+    return (
+      <div className="action-section">
+        <div className="action-property-title">Output directories</div>
+        {actionsResult.outputDirectories.length ? (
+          <div className="action-list">
+            {actionsResult.outputDirectories.map((dir) => (
+              <TreeNodeComponent
+                node={{
+                  obj: new build.bazel.remote.execution.v2.DirectoryNode({ name: dir.path, digest: dir.treeDigest }),
+                  type: "dir",
+                }}
+                treeShaToExpanded={this.state.treeShaToExpanded}
```

Comment:
Unlike input, which only has one tree, there could be multiple output directories in ActionResult, which is currently translated to multiple trees. This has aside effects:

The current implementation shares the same `treeShaToExpanded` state among all input/output trees. This could cause expanding 1 tree in input to also expand that tree in output. In practice, I think it's unlikely that these directories are duplicated. We could fix this later by providing a separate state for each tree.

Reply:
A second problem with this approach is that if there are 2 or more output dirs, the prefix `bazel-out/<target-platform>/bin` prefix is not duplicated between the rendered trees.

I briefly considered deduplicating them, but when we do that, it would not be clear to the user which directories in the merged tree were the output directory specified in the ActionResult.

I plan to add `Expected output paths` to render the Command data in a future PR. Once that's done, I think we could reconsider merging the output trees in our UI.